backend/tests/v4/integrated/signup.js

```diff
@@ -0,0 +1,624 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+/**
+ *  Copyright (C) 2014 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { expect } = require('chai');
+const app = require('../../../src/v4/services/api.js').createApp();
+
+const request = require('supertest');
+const C = require('../../../src/v4/constants');
+const responseCodes = require('../../../src/v4/response_codes.js');
+
+describe('Sign up', () => {
+	let server;
+
+	before((done) => {
+		server = app.listen(8080, () => {
+			console.log('API test server is listening on port 8080!');
+			done();
+		});
+	});
+
+	after((done) => {
+		server.close(() => {
+			console.log('API test server is closed');
+			done();
+		});
+	});
+
+	const username = 'signup_helloworld';
+	const uppercase_username = 'Signup_helloworld';
+	const password = 'Str0ngPassword!';
+	const email = 'test3drepo_signup@mailinator.com';
+	const firstName = 'Hello';
+	const lastName = 'World';
+	const countryCode = 'GB';
+	const company = 'company';
+	// const jobTitle = "someJobTitle";
```

Comment:
Yes, it's temporarily removed because we couldn't get the legal docs out in time. so we don't want to get rid of this as it'll be revived in the next release.